server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -138,3 +170,149 @@ func getOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Reque
+	if err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.create_connection.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	auditRec.Success()
+	auditRec.AddEventResultState(connection)
+	auditRec.AddEventObjectType("outgoing_oauth_connection")
+	c.LogAudit("client_id=" + connection.ClientId)
+
+	service.SanitizeConnection(connection)
+
+	w.WriteHeader(http.StatusCreated)
+	if err := json.NewEncoder(w).Encode(connection); err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.create_connection.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func updateOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Request) {
+	service, ok := ensureOutgoingOAuthConnectionInterface(c, whereOutgoingOAuthConnection)
+	if !ok {
+		return
+	}
+
+	c.RequireOutgoingOAuthConnectionId()
+	if c.Err != nil {
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("updateOutgoingOAuthConnection", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	audit.AddEventParameter(auditRec, "outgoing_oauth_connection_id", c.Params.OutgoingOAuthConnectionID)
+	c.LogAudit("attempt")
+
+	if !checkOutgoingOAuthConnectionWritePermissions(c) {
+		return
+	}
+
+	var inputConnection model.OutgoingOAuthConnection
+	if err := json.NewDecoder(r.Body).Decode(&inputConnection); err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.update_connection.input_error", nil, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if inputConnection.Id != c.Params.OutgoingOAuthConnectionID {
+		c.SetInvalidParam("id")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	currentConnection, err := service.GetConnection(c.AppContext, c.Params.OutgoingOAuthConnectionID)
+	if err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.update_connection.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	auditRec.AddEventPriorState(currentConnection)
+
+	connection, err := service.UpdateConnection(c.AppContext, &inputConnection)
+	if err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.update_connection.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	auditRec.AddEventObjectType("outgoing_oauth_connection")
+	auditRec.AddEventResultState(connection)
+	auditRec.Success()
+	auditLogExtraInfo := "success"
+	// Audit log changes to clientID/Client Secret
+	if connection.ClientId != currentConnection.ClientId {
+		auditLogExtraInfo += " new_client_id=" + connection.ClientId
+	}
+	if connection.ClientSecret != currentConnection.ClientSecret {
+		auditLogExtraInfo += " new_client_secret"
+	}
+	c.LogAudit(auditLogExtraInfo)
+	service.SanitizeConnection(connection)
+
+	w.WriteHeader(http.StatusOK)
```

Comment:
Is this needed? I'd expect the status code default to that in case of a successful write.